packages/consumption/src/modules/requests/outgoing/OutgoingRequestsController.ts

```diff
@@ -65,9 +85,13 @@ export class OutgoingRequestsController extends ConsumptionBaseController {
         }
 
         const innerResults = await this.canCreateItems(parsedParams.content, parsedParams.peer);
-
         const result = ValidationResult.fromItems(innerResults);
 
+        const keyUniquenessValidationResult = OutgoingRequestsController.validateKeyUniquenessOfRelationshipAttributesWithinRequest(parsedParams.content.items, parsedParams.peer);
```

Comment:
This can be moved into the `if (parsedParams.peer)`, then you don't need an optional recipient in all your new methods

Reply:
I think that this is unfortunately not possible, because even with an unknown `parsedParams.peer`, incorrect Requests can be created, which would lead to the creation of duplicate RelationshipAttributes with the same `key` regardless of the future recipient. This is because empty strings can also be specified as placeholders for potential recipients as `owner` of RelationshipAttributes.

Or am I missing something?

Reply:
No, I missed that you are checking the request items in isolation and don't need to know the relationship for that